dhcpv6/dhcpv6_test.go

```diff
@@ -82,72 +82,39 @@ func TestFromAndToBytes(t *testing.T) {
 	require.Equal(t, expected, toBytes)
 }
 
-func withServerID(d DHCPv6) DHCPv6 {
-	sid := OptServerId{}
-	d.AddOption(&sid)
-	return d
-}
-
 func TestNewAdvertiseFromSolicit(t *testing.T) {
 	s := DHCPv6Message{}
 	s.SetMessage(SOLICIT)
 	s.SetTransactionID(0xabcdef)
 	cid := OptClientId{}
 	s.AddOption(&cid)
+	duid := Duid{}
 
-	a, err := NewAdvertiseFromSolicit(&s, withServerID)
+	a, err := NewAdvertiseFromSolicit(&s, WithServerID(duid))
 	require.NoError(t, err)
 	require.Equal(t, a.(*DHCPv6Message).TransactionID(), s.TransactionID())
 	require.Equal(t, a.Type(), ADVERTISE)
 }
 
-func TestNewReplyFromRequest(t *testing.T) {
+func TestNewReplyFromDHCPv6Message(t *testing.T) {
 	req := DHCPv6Message{}
 	req.SetMessage(REQUEST)
```

Comment:
please also test the other cases (CONFIRM, RENEW, REBIND and RELEASE), and the negative test cases (e.g. SOLICIT and others)